File: backend/src/main/java/com/example/rentalcars/features/user/domain/port/inbound/UserService.java
package com.example.rentalcars.features.user.domain.port.inbound;

import com.example.rentalcars.features.user.infrastructure.adapter.inbound.rest.dto.UpdateUserRequest;
import com.example.rentalcars.features.user.infrastructure.adapter.inbound.rest.dto.UserRequest;
import com.example.rentalcars.features.user.domain.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;
import java.util.UUID;

public interface UserService {
    User getUserById(UUID id);
    User getUserByEmail(String email);
    User getInternalUserByEmail(String email);
    User register(UserRequest request);
    User update(UUID id, UpdateUserRequest request);
    Page<User> getAllUsers(Pageable pageable);
    void delete(UUID id);
    boolean existsByEmail(String email);
    Optional<User> findByResetToken(String token);
    void save(User user);
}

File: frontend/src/services/userService.js
import api from '../api/axios';

const userService = {

    getAllUsers: async (page = 0, size = 9) => {
        const response = await api.get(`/users?page=${page}&size=${size}`); 
        return response.data;
    },
    
    getUserById: async (id) => {
        const response = await api.get(`/users/${id}`);
        return response.data;
    }, 

    getUserByEmail: async (email) => {
        const response = await api.get(`/users/email/${email}`);
        return response.data;
    },
    
    deleteUser: async (id) => {
        const response = await api.delete(`/users/${id}`);
        return response.data;
    },

    updateUser: async (id, userData) => {
        const response = await api.put(`/users/${id}`, userData);
        return response.data;
    }
};

export default userService;